### DISCUSSION TITLE
Automatically named objects from *.11tydata.js files

### DISCUSSION THREAD

**Discussion by stuffnting** [11ty/eleventy#2996] [Q&A] [answered]
This might be a daft question, but I am stumped.

In the [official documentation](https://www.11ty.dev/docs/data-js/), there is a simple example of JavaScript Data Files.

```
module.exports = [
  "user1",
  "user2"
];
```

I can see that if this code was in a global data file within, let's say `_data/myData.js`, the array would be accessible in a template as `myData`. However, if the code is in a template data file, such as `posts/posts.11tydata.js`, how is the array accessible in a template?

I realise that I can just use this instead:

```
module.exports = { 
myData:
  [
    "user1",
    "user2"
  ]
};
```

But, is there no automatically named object for such data files?

**Comment by pdehaan** [comment]
I think the `myData` values should be available as `myData` within any templates in your posts/** directory since they should be getting similar to frontmatter values in your directory data file.

**Reply by pdehaan** [reply, marked as answer]
https://github.com/pdehaan/11ty-2996

```liquid
<!-- ./src/pages/index.liquid -->
global: <pre>{{ myData1 | json }}</pre>
local: <pre>{{ myData2 | json }}</pre>
```

```js
// ./src/_data/myData1.js
module.exports = [
  "global1",
  "global2"
];
```

```js
// ./src/pages/pages.11tydata.js
module.exports = {
  myData2: [
    "local1",
    "local2"
  ],
};
```

### OUTPUT
```html
global: <pre>["global1","global2"]</pre>
local: <pre>["local1","local2"]</pre>
```
